### DISCUSSION TITLE
more comments

### DISCUSSION THREAD

**Discussion by doki23** [databendlabs/databend#4866] [General]
We should add more comments for the code. 
It's not newcomer-friendly now 😞.

**Comment by Xuanwo** [comment]
Yep, it's a long-term issue and we need to take action now. Let's do something to change it instead.

Can you point out the most confusing part of mod/function/logic here? We can invite the code owner to add some docs as a first start.

**Reply by doki23** [reply]
Thanks for the attention. 
The most now confusing one for me is the *query* mod.

**Reply by Xuanwo** [reply]
> The most now confusing one for me is the _query_ mod.

Thanks for providing the information. But I'm afraid it's still not enough for us to take action or give an answer. `query` is a large mod that has 78k lines of code. How about starting by asking a quick, small question?

**Reply by doki23** [reply]
It's ok to me now, no need to solve small/concrete questions for me.
And I'm glad to contribute it when I'm familiar with the codebase.
